jackbot.py

```diff
@@ -66,11 +65,15 @@ async def _getRoles(self):
         roles = await self.ch.guild.fetch_roles()
         for role in roles:
             roleshash[role.name] = role
+        self.roleshash = roleshash
         return roleshash
     
-    def _generateMsg(self, merchant, role_name):
-        msg = "{} is selling {} in {} - ".format(merchant.name, role_name, merchant.region)
-        msg += "[{}]({})".format(merchant.zone, URL + merchant.zone_img)
+    def _generateMsg(self, merchant, embedLink=False):
+        msg = f"{merchant.name} is selling <@&{self.roleshash[merchant.card].id}> in {merchant.region} - "

```

Comment:
This may run into key error if server doesn't contain that role, for example if called by showStatus, as showStatus shows all cards currently available regardless if guild has the role.
also showStatus should probably update self.roleshash by calling _getRoles()

Reply:
Should be addressed now:
https://github.com/mayo320/jackbot/pull/1/commits/bbdf1b145e0753dc06aaabc091c276a2082ff1d8#diff-c3c4e3c1460ac3e98fa32e00e9526a448fe1a4ac77d92066ff3a55656169daddR74
https://github.com/mayo320/jackbot/pull/1/commits/bbdf1b145e0753dc06aaabc091c276a2082ff1d8#diff-c3c4e3c1460ac3e98fa32e00e9526a448fe1a4ac77d92066ff3a55656169daddR49